### ISSUE TITLE
Why don't you make a Github Pages website?

### ISSUE THREAD

**Issue by Kolahzary** [boyEstrogen/Anime-Girls-Holding-Programming-Books#56]
You do well, but I think it would be better if combined with a simple html

**Comment by ishidex2** [comment]
It'd be good to make this page access through github api's to get all files and not copypasting

**Comment by Asday** [comment]
Off-topic, but are you _the_ Shadman?

**Comment by boyEstrogen** [comment]
Do you mean like some kind web page where you could view the images more easily.

If thats what you mean I would definitely be open to doing that, and adding it as project for the repo.

**Comment by reaysawa** [comment]
I've been thinking of doing something along those lines: having a sidebar with the name of the girls and a "selection screen" beside it that displays the character, similar to how the character selection pans out in fighting games. I think it wouldn't be too difficult to make something presentable.

If going on that direction, I can think of a few good libraries that enable this:
https://github.com/element-motion/element-motion
https://www.react-spring.io/

And aside from that, a carousel that fills the remaining of the screen. That's it.

**Comment by Asday** [comment]
As a professional web developer, heck c*rousels.

**Comment by reaysawa** [comment]
True! It could be a long container with vertical scrolling; that would make it easier to download everything at once... Not that carousels make it difficult. It's just the first thing that popped out on me because I'm thinking of fighting games' costume selection.

I'll see if I can get something going tomorrow.

**Comment by reaysawa** [comment]
For the record I did try to start, but I'm so out of touch with front-end development that I ended up spiralling into depressive self-unworthiness (this actually happened). I can't go as fast as I could, thinking of modelling components and such, so it's a hurdle. If I do get back on my feet at front-end, I'll try once again.

**Comment by Asday** [comment]
Sounds like you're overthinking it.  Pick a Hugo gallery template, tweak it a bit, and you're done.

**Comment by Xetera** [comment]
I'm interested in working on this so I made a rough mockup. The colors and overall layout will probably change but this is as far as I can push my creative boundaries at 9am.

<img width="1440" alt="ghpb" src="https://user-images.githubusercontent.com/24978328/60771472-797c2700-a09d-11e9-9c1e-c6dbba654c43.png">

There are certain images that are extremely off center so we might need to keep some configuration files
for problematic images to center them manually. It's not a huge deal though, as we might already want to keep metadata of images like what character is in the picture or what anime it's from to avoid this meme

![](https://pa1.narvii.com/6234/55a48a5733df5977e78ef0a70aaa608313f920f7_hq.gif)

Let me know what you think, happy to do tweaks or start working on the real thing. It would most likely be in [gatsby](https://github.com/gatsbyjs/gatsby) to address problems like gigantic image sizes and loading of the metadata I mentioned earlier. We could even go as far as to offer zip files of the entire archive if that's something you would be interested in.

**Comment by boyEstrogen** [comment]
that looks really good. do you think we should start a separate repo for the pages source or would cramming it into a folder on this repo be better

**Comment by Xetera** [comment]
It's probably better to keep everything in one place because this repo will be the "one source of truth" so just making a `web` folder or something would probably work better for this. I can get a basic preview going on netlify for my own fork and if you're happy with the overall layout you can make a separate branch and I'll PR it :+1:

**Comment by bucket3432** [comment]
We can always stuff it into the `gh-pages` branch if it can be deployed directly or another branch if it can't, but I agree with keeping it in this repo. I wouldn't want it in the `master` branch, though.

**Comment by bucket3432** [comment]
> Also I noticed that you guys are starting to run out of space
@Xetera where do you see that we're running out of space? The repo is only at 88MB, which is way under the recommended 1GB limit (and the hard limit is 100GB, which is even further off).

**Comment by Asday** [comment]
That's not what branches are for though...

I don't see why you wouldn't want `master` to contain the website.  It would stay up to date with new pictures much more easily that way, and all it takes is one `index.html` at the root or in `docs/`.

**Comment by bucket3432** [comment]
That's not what branches are commonly used for, but it's still a valid use of a branch. It's perfectly valid to say that all Web-related code is developed on another branch. One might argue that that's a reason for it to be in another repo, and that's also a perfectly fine option as well. The only argument I have for it being in the same repo is that it's tightly coupled with this repo; if the site is general enough that it's only loosely coupled, then it _should_ go in a different repo.

I'd rather keep it out of `master` because that's not really the deliverable of this repo, _unless_ we say that the site can be run locally and is the preferred way to view the images in the repo.

**Comment by Asday** [comment]
What's your definition of "valid use"?  Sure it would _work_, but it would be a lot more work to maintain.  Why make that harder?

I'm speaking from the experience of dealing with `redux-form`, where the website is stored on a different branch.  It's certainly not how it should be done if you expect people to be using github in relation to your repo, which, being on github, we do.

**Comment by Xetera** [comment]
@bucket3432 oh whoops, I guess I misread the earlier conversation. In that case, it doesn't really matter if the images themselves are large or not. 

I was thinking it would be more convenient to deploy the website on Netlify. There wouldn't need to be any html file involved in the repo at all, just the source code for the website. The repo I have now with a semi-working website is just the exact setup you guys have in this repo except with a `website` folder in the root. I don't think that would be problematic, would it? I don't think there's any need to loosely couple anything since the website is specifically only for this repo.

**Comment by Asday** [comment]
If you literally rename `website/` to `docs/`, then it'll work fine on github pages.  No need to add on some third party service.

**Comment by Xetera** [comment]
That works too, but then we'll need travis or circle CI to compile the react code into html and push to a new branch. It's also possible to do it with pure html and js and not bother with a static site generator but then we'd be serving 12M of images for C and 31M for SICP, I don't think anyone would like that. I'm fine with setting up a simple build step with Travis CI, it already supports this kind of thing by default but there's a tiny bit of configuration to be done on the repo owner's end.

**Comment by Asday** [comment]
I didn't think about thumbnailing, you are right.  Shame there's no good service for that.

**Comment by bucket3432** [comment]
> if you expect people to be using github

Alright, I concede the point in the context of GitHub. In that case, I don't have any strong reasons for or against either way.

As for thumbnailing, we can probably run something on the CI service to resize the image, or find a good CDN that will do it for us.

**Comment by Asday** [comment]
"Good, free, pick one."

I could run a service on my home server, but then my internet goes out every so often, and I'm moving house soon.

I don't imagine anyone here's looking to pay, either.

**Comment by Xetera** [comment]
Netlify lets you do all of this in about 4 clicks, it's also free. The resizing is taken care of by gatsby so you don't have to worry about a cdn or any other CI tool for that. I've just tested it on my end and the 12Mb C folder is shrunk all the way down to 400kb with full quality 250x250 webp previews :+1: 

As for netlify, you just sign in with github, click on the repo and it will detect the netlify configuration in the repo and figure out how to build and deploy the website on its own, it's really convenient.

**Comment by bucket3432** [comment]
This repo doesn't get updates too often, so I don't expect long build times to be an issue. I'd be more worried about how someone can test their changes locally.

**Comment by Asday** [comment]
Netlify supports hugo too, right?  Hugo's build times are measured in milliseconds, from my (admittedly limited) experience.

**Comment by Xetera** [comment]
I'll have instructions for all that in the `website` folder. A single `npm run develop` should be enough to get the website up locally if you ever want to test it. Netlify also creates previews of the new website when pull requests are created so when new images are created you should be able to see what the new website will look like.

From what I know, when you introduce tooling to Hugo for processing css and minifying assets and such, the build times are comparable to Gatsby's. I haven't looked into Hugo much, I just prefer Gatsby because React is fantastic :ok_hand:

**Comment by Xetera** [comment]
I have a basic preview up on https://anime-girls-holding-programming-books.netlify.com it's not styled yet but it's got the basic functionality for now

**Comment by bake** [comment]
I've never used Gatsby or committed to this repo so this might not be useful at all, but I'm used to separate theme and content in different repositories. With Hugo, for example, you'd create a single config file containing metadata about the page itself (like title or theme) and markdown files with arbitrary metadata about the images (could be used for the character, source, ...). This way we wouldn't have to have everything in this (currently only images containing) repository. Is something similar possible with Gatsby?

As a side note: [Hugo comes with image processing](https://gohugo.io/content-management/image-processing/), though it might not look as good as your implementation.

**Comment by shrekuu** [comment]
WOWOWOWOW!!! EXCITED!!! I am going to make one too!!!!! On it!!!
